pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -93,13 +94,54 @@ type LoadBalancerOpts struct {
 	MonitorMaxRetries uint       `gcfg:"monitor-max-retries"`
 }
 
+type OpenStackInstance struct {
+	InstanceID       string `json:"uuid"`
+	Name             string `json:"name"`
+	Hostname         string `json:"hostname"`
+	AvailabilityZone string `json:"availability-zone"`
+}
+
+// Network sets up the OpenStack network endpoint.
+func (os *OpenStack) Network() error {
+	if os.network != nil {
+		return nil
+	}
+	network, err := openstack.NewNetworkV2(os.provider, gophercloud.EndpointOpts{
+		Region: os.region,
+	})
+	if err != nil {
+		glog.Warningf("Failed to find neutron endpoint: %v", err)
```

Comment:
nit: This log message uses the service name (`neutron`) whereas the one below uses the service type (`compute`). Perhaps this message could use `network` instead.